catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/HistoryMetacardImpl.java

```diff
@@ -0,0 +1,249 @@
+                BasicTypes.STRING_TYPE));
+        historyDescriptors.add(new AttributeDescriptorImpl(EDITED_BY,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                false /* multivalued */,
+                BasicTypes.STRING_TYPE));
+        historyDescriptors.add(new AttributeDescriptorImpl(VERSIONED,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                false /* multivalued */,
+                BasicTypes.DATE_TYPE));
+        historyDescriptors.add(new AttributeDescriptorImpl(ID_HISTORY,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                false /* multivalued */,
+                BasicTypes.STRING_TYPE));
+        historyDescriptors.add(new AttributeDescriptorImpl(TAGS_HISTORY,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                true /* multivalued */,
+                BasicTypes.STRING_TYPE));
+
+        versionHistoryMetacard = new MetacardTypeImpl(METACARD_TYPE_NAME, historyDescriptors);
+    }
+
+    /**
+     * Will convert the given {@link Metacard} to a {@link HistoryMetacardImpl} by cloning
+     * it and adding the current subject, time, and a random UUID. Cannot take a
+     * {@link HistoryMetacardImpl} as the sourceMetacard.
+     *
+     * @param sourceMetacard Metacard to clone and create a history item from
+     * @param action         Which action was done to modify the metacard
+     * @throws IllegalArgumentException
+     */
+    public HistoryMetacardImpl(Metacard sourceMetacard, Action action, Subject subject) {
+        super(sourceMetacard, versionHistoryMetacard);
+        if (sourceMetacard instanceof HistoryMetacardImpl) {
+            throw new IllegalArgumentException(
+                    "Cannot create a history item from a history metacard.");
+        }
+
+        this.setState(action);
+        this.setIdHistory(sourceMetacard.getId());
+        this.setTagsHistory(sourceMetacard.getTags());
+
+        String editedBy = SubjectUtils.getEmailAddress(subject);
+        if (editedBy == null || editedBy.equals("")) {
+            editedBy = SubjectUtils.getName(subject);
+        }
+        this.setEditedBy(editedBy);
+
+        this.setVersioned(Date.from(Instant.now()));
+        this.setId(UUID.randomUUID()
+                .toString()
+                .replace("-", ""));
+        this.setTags(Collections.singleton(HISTORY_TAG));
+    }
+
+    /**
+     * Returns a {@link BasicTypes#BASIC_METACARD} version of the given
+     * {@link HistoryMetacardImpl}
+     *
+     * @return The converted metacard
+     * @throws IllegalStateException
+     */
+    public Metacard toBasicMetacard() {
+        String id = this.getIdHistory();
+        if (id == null || id.matches("\\s*")) {
```

Comment:
Apache's `StringUtils` specifically would be great for this as it returns true for nulls or any whitespace/empty strings.